addons/hr_expense/models/hr_expense_sheet.py

```diff
@@ -294,6 +301,12 @@ def _compute_state(self):
             else:
                 sheet.state = 'draft'
 
+    @api.depends('expense_line_ids.attachment_ids')
+    def _compute_main_attachment(self):
+        for sheet in self:
+            if not sheet.message_main_attachment_id or sheet.message_main_attachment_id not in sheet.attachment_ids:
+                sheet.message_main_attachment_id = sheet.attachment_ids[:1]
+
```

Comment:
```suggestion
    @api.depends('attachment_ids')
    def _compute_main_attachment(self):
        for sheet in self:
            attachments = sheet.attachment_ids
            if not sheet.message_main_attachment_id or sheet.message_main_attachment_id not in attachments:
                expenses = sheet.expense_line_ids
                new_mma_checksums = expenses.message_main_attachment_id.mapped('checksum')
                sheet.message_main_attachment_id = \
                    attachments.filtered(lambda att: att.checksum in new_mma_checksums)[:1] or attachments[:1]
```
What about? Seems good enough

Reply:
Ow, I didn't think of that, tested it, seems to work fine, so I've pushed your suggestion.
I didn't add a compute just to set the main_attachment of an expense, I think it's not a "necessity".
I've updated the commit msg to reflect the change.
If you think it's good let me know and I will continue with the fw-porting :)